### PR TITLE
Feat/Snippet Storage Setup for #107

### PR DESCRIPTION
**Pull request by ThijsZijdel**
Initial Setup Complete

- Core snippets structure has been established.
- Basic functionality is in place.

### UI/UX refinement, including:
- [ ] Implementing tag management for better organization.
- [ ] Improving folder structures, such as paths/ for clarity and consistency.
- [ ] Adding functionality for file annotations/notes for better context.
- [ ] Implement search.

### Code Refactor, including:
- [ ] Refactor Update Functions
- [ ] Debounce db updates
- [ ] Move out Rust sqlite migrations

### Feature expansions 
- [ ] Sharing?
- [ ] 

_Data structures are mostly done for these_

### CODE REVIEW

File: src/utils/database.ts
Comment (Sparkenstein) [inline src/utils/database.ts:25]: Does this handle escape sequences in the code? 
for example will it store following snippet:

```js
const str = "he said \"don't go there\""
```

or a few other test cases, combination of single quotes, double, tripple (python), double escape (escape of escape)

File: src-tauri/src/main.rs
Comment (Sparkenstein) [inline src-tauri/src/main.rs:30]: this needs to be `IF NOT EXISTS`

File: src/Features/colors/CustomPicker.tsx
Comment (Sparkenstein) [inline src/Features/colors/CustomPicker.tsx:168]: my bad

**Review by Sparkenstein — APPROVED** [review]
LGTM, mind if I do some changes after merging? I will be primarily optimizing queries, setting up some structure, might install some ORM as well not sure. because I was thinking of saving some settings to DB as well, like JSON editor or pastebins etc.

**Comment by ThijsZijdel** [comment]
> LGTM, mind if I do some changes after merging? I will be primarily optimizing queries, setting up some structure, might install some ORM as well not sure. because I was thinking of saving some settings to DB as well, like JSON editor or pastebins etc. 

Yeah for sure! Go ahead, missed the 'IF NOT EXISTS' and the setup was a rough draft. But an ORM could definitely make it easier to extend. 

Lemme know when you're ready. Will be continuing on the other feature's ✌️

**Pull request merged**
